src/luarocks/cmd.lua

```diff
@@ -322,69 +429,43 @@ function cmd.run_command(description, commands, external_namespace, ...)
             end
          end
       end
-      local nonflags = { util.parse_flags(unpack(args)) }
-      local flags = table.remove(nonflags, 1)
-      if flags.ERROR then
-         die(flags.ERROR.." See --help.")
-      end
-
-      -- Compatibility for old names of some flags
-      if flags["to"] then flags["tree"] = flags["to"] end
-      if flags["from"] then flags["server"] = flags["from"] end
-      if flags["nodeps"] then flags["deps-mode"] = "none" end
-      if flags["only-from"] then flags["only-server"] = flags["only-from"] end
-      if flags["only-sources-from"] then flags["only-sources"] = flags["only-sources-from"] end
 
-      return flags, nonflags, cmdline_vars
+      return args, cmdline_vars
    end
 
-   local flags, nonflags, cmdline_vars = process_arguments(...)
+   local args, cmdline_vars = process_cmdline_vars(...)
+   local parser = get_parser(program_name, description, cmd_modules)
+   args = parser:parse(args)
 
-   if flags["timeout"] then   -- setting it in the config file will kick-in earlier in the process
-      local timeout = tonumber(flags["timeout"])
-      if timeout then
-         cfg.connection_timeout = timeout
-      else
-         die "Argument error: --timeout expects a numeric argument."
-      end
+   if not args.command then
+      util.printout(parser:get_help())
+      os.exit(cmd.errorcodes.OK)
    end
 
-   local command
-   if flags["help"] or #nonflags == 0 then
-      command = "help"
-   else
-      command = table.remove(nonflags, 1)
+   -- Compatibility for old flag
+   if args.nodeps then
+      args.deps_mode = "none"
```

Comment:
Can this be done with argparse aliases?

Reply:
I don't think so since `--nodeps` is a flag while `--deps-mode` takes an argument.